changes.go

```diff
@@ -212,7 +212,7 @@ func (ch *AddServiceChange) GUIArgs() []interface{} {
 	if options == nil {
 		options = make(map[string]interface{}, 0)
 	}
-	return []interface{}{ch.Params.Charm, ch.Params.Service, options}
+	return []interface{}{ch.Params.Charm, ch.Params.Service, options, ch.Params.Constraints}
```

Comment:
We treat GUIArgs as positional and we always return all the args, even if they are null/unset values.